client/proposal.go

```diff
@@ -181,9 +190,19 @@ func (c *Client) ProposeChannel(ctx context.Context, prop ChannelProposal) (*Cha
 		return nil, errors.WithMessage(err, "channel proposal")
 	}
 
-	// 3. fund
-	fundingErr := c.fundChannel(ctx, ch, prop)
-	return ch, fundingErr
+	// 3. fund the channel. If funding fails, settle the channel.
+	if err := c.fundChannel(ctx, ch, prop); err != nil {
+		c.log.WithField("channel", ch.ID()).Warn("Funding failed, will settle channel.")
+		// Increment the account usage, or we will not be able to settle since
+		// the funding failed.
+		c.wallet.IncrementUsage(ch.Params().Parts[ch.machine.Idx()])
```

Comment:
Okay, I see but then `wallet.Increment` would be called twice if the channel would be funded correctly, namely in [`completeFunding`](https://github.com/hyperledger-labs/go-perun/blob/main/client/proposal.go#L652).
Maybe we can quickly talk about this later.

Reply:
Of course you need to remove it there...

Reply:
Yes true, even though this would contradict the usage of `wallet.Increment` a bit because the account hasn't been "used" unless the funding really completes. But this is probably a cleaner approach. I'll change it.

Reply:
You can check the completeCPP function. In fact, the account is unlocked there, so one might argue that it is even more sensible to increment the usage there (and not in completeFunding).